src/elevator/fsm.rs

```diff
@@ -203,7 +220,26 @@ impl ElevatorFSM {
                                 self.close_door();
                             }
                         }
-                        Moving => (), // Should implement stop button logic here
+                        Moving => {
+                            if !self.started_moving {
+                                self.motor_timer = Instant::now() + Duration::from_millis(self.motor_driving_time);
+                                self.started_moving = true;
+                            }
+
+                            if self.motor_timer <= Instant::now() && self.started_moving {
```

Comment:
Do we need to "and" this with self.started_moving? I don't think self.started_moving should ever be false when it's inside the "Moving" block.